repl/src/main/scala/ammonite/repl/interp/AmmonitePlugin.scala

```diff
@@ -150,4 +170,49 @@ object AmmonitePlugin{
 
     output(open.toVector)
   }
+}
+
+
+object LineNumberModifier {
+  def apply(g: Global)(unit: g.CompilationUnit,
+                       output: Seq[ImportData] => Unit,
+                       importsLen: => Int) = {
+
+    import g._
+
+    class LineNumberCorrector(unit: CompilationUnit) extends Transformer {
+      override def transform(tree: Tree) = {
+        val transformedTree = super.transform(tree)
+        tree.pos match {
+          case s: scala.reflect.internal.util.OffsetPosition =>
```

Comment:
Let's import `scala.reflect.internal.util.{OffsetPosition, BatchSourceFile}` at the top so we can keep our code a bit more concise